src/cascadia/TerminalApp/CascadiaSettings.cpp

```diff
@@ -452,3 +452,145 @@ void CascadiaSettings::_ValidateAllSchemesExist()
+                                                                   const NewTerminalArgs& newTerminalArgs) const
+{
+    const GUID profileGuid = _GetProfileForIndexAndArgs(profileIndex, newTerminalArgs);
+    auto settings = MakeSettings(profileGuid);
+
+    if (newTerminalArgs)
+    {
+        // Override commandline, starting directory if they exist in newTerminalArgs
+        if (!newTerminalArgs.Commandline().empty())
+        {
+            settings.Commandline(newTerminalArgs.Commandline());
+        }
+        if (!newTerminalArgs.StartingDirectory().empty())
+        {
+            settings.StartingDirectory(newTerminalArgs.StartingDirectory());
+        }
+        if (!newTerminalArgs.TabTitle().empty())
+        {
+            settings.StartingTitle(newTerminalArgs.TabTitle());
+        }
+    }
+
+    return { profileGuid, settings };
+}
+
+// Method Description:
+// - Helper to get the GUID of a profile, given an optional index and a possible
+//   "profile" value to override that.
+// - First, we'll try looking up the profile for the given index. This will
+//   either get us the GUID of the Nth profile, or the GUID of the default
+//   profile.
+// - Then, if there was a Profile set in the NewTerminalArgs, we'll use that to
+//   try and look the profile up by either GUID or name.
+// Arguments:
+// - index: if provided, the index in the list of profiles to get the GUID for.
+//   If omitted, instead use the default profile's GUID
+// - newTerminalArgs: An object that may contain a profile name or GUID to
+//   actually use. If the Profile value is not a guid, we'll treat it as a name,
+//   and attempt to look the profile up by name instead.
+// Return Value:
+// - the GUID of the profile corresponding to this combination of index and NewTerminalArgs
+GUID CascadiaSettings::_GetProfileForIndexAndArgs(std::optional<int> index,
+                                                  const NewTerminalArgs& newTerminalArgs) const
+{
+    GUID profileGuid = _GetProfileForIndex(index);
+
+    if (newTerminalArgs)
+    {
+        // First, try and parse the "profile" argument as a GUID. If it's a
+        // GUID, and the GUID of one of our profiles, then use that as the
+        // profile GUID instead. If it's not, then try looking it up as a
+        // name of a profile. If it's still not that, then just ignore it.
+        if (!newTerminalArgs.Profile().empty())
+        {
+            bool wasGuid = false;
+
+            try
+            {
+                const auto newGUID = Utils::GuidFromString(newTerminalArgs.Profile().c_str());
+
+                for (const auto& p : _profiles)
+                {
+                    if (p.GetGuid() == newGUID)
+                    {
+                        profileGuid = newGUID;
+                        wasGuid = true;
+                        break;
+                    }
+                }
+            }
+            CATCH_LOG();

```

Comment:
probably we don't want to log when this happens -- we should check it heuristically first?